### DISCUSSION TITLE
Middleware - number of results

### DISCUSSION THREAD

**Discussion by pixiitech** [algolia/instantsearch#5921] [Q&A]
I'm trying to add a middleware for analytics use, based on this code example: https://www.algolia.com/doc/api-reference/widgets/middleware/react/#examples

I'm required to get the number of results but I'm only offered `uiState` which only has the `refinementList` and `query`. How can I get more information on the status of the search??

**Comment by pixiitech** [comment]
I couldn't get what I needed using insights, created a component rendered on the page that sent the data upon rendering to a debounced function.

**Comment by Haroenv** [comment]
You can use the `instantSearchInstance.mainIndex.getResults()`, but a widget also works (that's how our previous analytics widget worked). Middleware isn't designed today to read results, as most use cases would require rendering, and that calls for a widget.
